### ISSUE TITLE
Datepicker component emits an input event, each time the model is changed

### ISSUE THREAD

**Issue by zedrdave** [buefy/buefy#1361]
### Overview of the problem

**Buefy** version: 0.7.5
**Vuejs** version: 2.6.10
**OS/Browser**: OS X (Mojave)

### Description

The `Datepicker` component emits an `input` event when the value of `dateSelected` changes (through a `watch`). This leads to unwanted cascade triggers and potential infinite loops, when the component is updated through its model.

It seems the solution would be to emit the events in the `onChange`/`onChangeNative` methods instead. 

### Steps to reproduce

```
<template>
<div>
    <b-checkbox v-model="item" />
    <b-checkbox v-model="item" />
</div>
</template>
<script>
export default {
  name: 'TestDatepicker',
  data: function() {
    return {
      fakeState: new Date()
    }
  },
  computed: {
    item: {
      get() {
        return this.fakeState;
      },
      set(val) {
        console.log('set: ' + val);
        this.fakeState = val;
      }
    },
  },
}
```
### Expected behavior

When manually picking a date in either picker, `item.set()` should be called only once.

### Actual behavior

An `input` event is triggered on the second picker and `item.set()` is called a second time.
If the model is connected to state mutation that changes the value, this leads to an infinite loop.

**Note:** Timepicker is likely affected as well.

**Comment by zedrdave** [comment]
Any idea on a way to work around this in the meantime?

**Comment by jtommy** [comment]
Fix https://github.com/buefy/buefy/commit/98c736a719dce146a0e9fe1e991655633a6292aa

**Comment by zedrdave** [comment]
Thanks for the fix!
BTW: I think Timepicker has a similar issue. Should I log a separate issue?

**Comment by jtommy** [comment]
I'll fix it later, don't worry!
